### PR TITLE
Replace react-select with Material-UI's Autocomplete

### PR DESCRIPTION
**Pull request by KarboniteKream**
### Motivation
- In effort to reduce the JS bundle size, we can use Material-UI components instead of extra dependencies

### Modifications
- Replace `react-select` with Material-UI Lab's `Autocomplete` component.
- Currently selected option no longer disappears when menu is open. The text is selected instead.
- Remove the item limit. The dropdown works fine with 5,000 items.

Before | After
:------:|:-----:
before-small | after-small
before-selected | after-overflow
before-normal | after-normal

### Result
- Improve appearance and UX, with less code
- Remove a dependency and reduce `docs-client` bundle size by 70 kB (down to 572 kB)

### CODE REVIEW

**Comment by KarboniteKream** [comment]
I tried keeping the CSS modifications as minimal as possible, while keeping the old appearance.

Hiding the currently selected item when the menu is open would be a bit more difficult with `Autoselect`, since the implementation is different. It now allows the user to quickly edit the text, while also clearly indicating that you can start typing again (which was the goal of #3095).

I also removed the limit of the displayed items. I tested with 5,000 items and it works fine. There is a way to configure a limit, which I will look into tomorrow, but please let me know if you prefer the limit or not.

I suggest you test this PR locally to get a better idea of the changes.

**Comment by ikhoon** [comment]
This looks good! ♥️ Let me review and test it soon.

**Review by trustin — APPROVED** [review]
Niiiice! 🚀

**Review by ikhoon — APPROVED** [review]
Nice work! @KarboniteKream 🙇‍♂️

File: docs-client/src/components/GotoSelect/index.tsx [hunk 1]
```diff
@@ -14,417 +14,136 @@
...
-
-function IndicatorsContainer(props: IndicatorContainerProps<OptionType>) {
-  return (
-    <div className={props.selectProps.classes.indicatorsContainer}>
-      {props.children}
-    </div>
-  );
-}
+const filterOptions = createFilterOptions<Option>({
+  limit: 500,
```
Comment (ikhoon) [inline docs-client/src/components/GotoSelect/index.tsx:54]: 👍

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Pull request merged**